File: javascript/cluster_management/src/delete-multi-region.js
import { DSQLClient } from "@aws-sdk/client-dsql";
import { DeleteMultiRegionClustersCommand } from "@aws-sdk/client-dsql";

async function deleteMultiRegionClusters(linkedClusterArns, client) {
    const deleteMultiRegionClustersCommand = new DeleteMultiRegionClustersCommand({
        linkedClusterArns: linkedClusterArns,
    });
    try {
        const response = await client.send(deleteMultiRegionClustersCommand);
        return response;
    } catch (error) {
        if (error.name === "ResourceNotFoundException") {
            console.log("Some or all Cluster ARNs not found or already deleted");
        } else {
            console.error("Unable to delete multi-region clusters: ", error.message);
        }
        throw error;
    }
}

async function main() {
    const region = "us-east-1";
    const client = new DSQLClient({ region });
    const linkedClusterArns = [
        "arn:aws:dsql:us-east-1:111111999999::cluster/foo0bar1baz2quux3quuux4",
        "arn:aws:dsql:us-east-2:111111999999::cluster/bar0foo1baz2quux3quuux4"
    ];

    const response = await deleteMultiRegionClusters(linkedClusterArns, client);
    console.log("Deleting Clusters with ARNs:", linkedClusterArns);
}

main();

File: python/cluster_management/src/delete_multi_region.py
import boto3

def delete_multi_region_clusters(linkedClusterArns, client):
    client.delete_multi_region_clusters(linkedClusterArns=linkedClusterArns)

def main():
    region = "us-east-1"
    client = boto3.client("dsql", region_name=region)
    linkedClusterArns = [
        "arn:aws:dsql:us-east-1:111111999999::cluster/foo0bar1baz2quux3quuux4",
        "arn:aws:dsql:us-east-2:111111999999::cluster/bar0foo1baz2quux3quuux4"
    ]
    delete_multi_region_clusters(linkedClusterArns, client)
    print("Deleting clusters with ARNs:", linkedClusterArns)


if __name__ == "__main__":
    main()
